### PR TITLE
Various RCU related update

### PR DESCRIPTION
**Pull request by jshum2479**
This PR ccovers 

* Remove he use of getDatabaseDefaults when creating JRF domain
* Deprecate the user of several atp.* and ssl.* fields in RCUDbInfo 
* Added documentation how to support multi data source, AGL in sparse model
* Refactoring

### CODE REVIEW

File: documentation/2.0/content/rcuinfo.md [hunk 1]
```diff
@@ -1,10 +1,31 @@
 ### Specifying RCU connection information in the model
 
-During creating or updating a domain, there is new section `RCUDbInfo` in the model, under the `domainInfo` section, for specifying RCU database connection information without using the command-line arguments `-rcu_db` and `-rcu_prefix`. Use this to support a database where the connection string is more complex and requires extra options.
+When creating a JRF domain, you can provide all the RCU connection information in a section `RCUDbInfo` under the `domainInfo` section in the model.  
+It provides you with more flexibility over the basic command line arguments of
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:4]: command line -> command-line

File: documentation/2.0/content/rcuinfo.md [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+| Data source name        | JNDI name                  | Schema  | Target |
+|-------------------------|----------------------------|---------|---|
+| WLSSchemaDataSource     | jdbc/WLSSchemaDataSource   | prefix_WLS_RUNTIME | None |
+| LocalSvcTblDataSource   | jdbc/WLSSchemaDataSource   | prefix_STB | admin server |
+| opss-data-source        | jdbc/OpssDataSource        | prefix_OPSS | admin server and cluster |
+| opss-audit-viewDS       | jdbc/AuditViewDataSource   | prefix_IAU_VIEWER | admin server and cluster |
+| opss-audit-DBDS         | jdbc/AuditAppendDataSource | prefix_IAU_APPEND | admin server and cluster |
+| mds-owsm                | jdbc/mds/owsm              | prefix_MDS | admin server and cluster |
+
+By default, the JRF domain template data source only has the default information such as URL and schema with the default prefix `DEV`.  During domain creation,
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:20]: only has -> has only

File: documentation/2.0/content/rcuinfo.md [hunk 3]
```diff
@@ -1,10 +1,31 @@
...
+WDT will use the information you provided in the command line or in the `RCUDbinfo` section to override the default values from the template so that it can connect to the database you specified.
+
+For some advanced use case, such as using Oracle Active Grid Link data source or Multi data sources, you can provide a sparse model of the above data sources in a separate model file
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:23]: advanced use case -> advanced use cases
using Oracle -> using an Oracle
Active Grid Link -> Active GridLink data source
Multi data sources -> Multi Data Sources
of the above data sources -> of the data sources

File: documentation/2.0/content/rcuinfo.md [hunk 4]
```diff
@@ -1,10 +1,31 @@
...
+during domain creation.  See [Advance use cases](#advanced-jrf-database-use-cases)
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:24]: Period at the end of the sentence.

File: documentation/2.0/content/rcuinfo.md [hunk 5]
```diff
@@ -1,10 +1,31 @@
...
 
 ### Access a database using a wallet
 
-When accessing a database using a wallet, you need to obtain the wallet from your DBA and information about the database:
+When accessing a database such as ATP or SSL using a wallet, you need to obtain the wallet from your DBA and information about the database:
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:28]: database such as ATP or SSL -> database, such as ATP or SSL, (two commas needed)

File: documentation/2.0/content/rcuinfo.md [hunk 6]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
         storageXMLLocation: wlsdeploy/rcu/config/MyStorage.xml
 ```
+
+### Advanced JRF database use cases
+
+In the following examples of the JRF data source sparse model, you can use it to further customize the JRF domain template data sources.
+
+#### Default template data source
+
+This is a sparse model for JRF data sources with the rcu prefix `FMW1`.
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:192]: rcu  -> RCU

File: documentation/2.0/content/rcuinfo.md [hunk 7]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+
+### Advanced JRF database use cases
+
+In the following examples of the JRF data source sparse model, you can use it to further customize the JRF domain template data sources.
+
+#### Default template data source
+
+This is a sparse model for JRF data sources with the rcu prefix `FMW1`.  
+You will need to update at least the `URL`, `PasswordEncrypted`, and the `user` property value.  When you specify the value of `URL`, it 
+must be a valid `JDBC URL` format, which is different from the `rcu_db_conn_string` which can do not require the `jdbc:oracle:thin:...` part.
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:194]: the `rcu_db_conn_string` which can do not require -> the `rcu_db_conn_string`, which does not require

File: documentation/2.0/content/rcuinfo.md [hunk 8]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+                        oracle.net.CONNECT_TIMEOUT:
+                            Value: '120000'
+                        SendStreamAsBlob:
+                            Value: 'true'
+```
+
+
+#### Oracle Active Grid Link Data Source (AGL)
+
+For setting data source to access Oracle Active Grid Link database, besides updating the `URL`, `PasswordEncrypted`, and the `user` property value;
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:316]: setting data source -> setting the data source

File: documentation/2.0/content/rcuinfo.md [hunk 9]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+```yaml
+      mds-owsm:
+            Target: cluster-1,admin-server
+            JdbcResource:
+              JDBCOracleParams:
+                OnsNodeList:  'node1, node2'
+                ....
+```
+
+You can run WDT tool `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParmas` for the complete list
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:328]: re-write: For the complete list of fields, run the WDT command, `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParmas`.

File: documentation/2.0/content/rcuinfo.md [hunk 10]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+                OnsNodeList:  'node1, node2'
+                ....
+```
+
+You can run WDT tool `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParmas` for the complete list 
+of fields.
+
+#### Multi Data Sources
+
+If your database does not support high availability natively such as RAC or AGL, then WebLogic provides a Multi Data Source option, which basically the main data source is a logical data source that
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:333]: natively such -> natively, such
which basically -> where basically

File: documentation/2.0/content/rcuinfo.md [hunk 11]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+contains multiple physical data sources.   In order to create a spare model,  you can for each data source
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:334]: you can for each data source -> for each data source, you can:

File: documentation/2.0/content/rcuinfo.md [hunk 12]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+
+1. Update the templated data source `URL`, `EncryptedPassword`, and 'user' property value.  In the `URL` part, you can specify one of the accessible physical database URL.
+2. Duplicate each datasource two times
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:337]: Period at the end of the sentence

File: documentation/2.0/content/rcuinfo.md [hunk 13]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+3. Change the duplicated data source name, `JNDIName` by appending `-1`, `-2` etc. The `-1`, `-2` data sources are the physical data source, you will need to update their `URL` too.
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:338]: etc -> , and such.
data source, you will -> data sources; (semi-colon) you will

File: documentation/2.0/content/rcuinfo.md [hunk 14]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+
+For example (note: details replaced by `....` for easier reading),
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:340]: reading), -> reading): (colon)

File: documentation/2.0/content/rcuinfo.md [hunk 15]
```diff
@@ -161,3 +182,205 @@ domainInfo:
...
+```yaml
+       opss-data-source:
+           ....
+            JDBCDataSourceParams:
+              DataSourceList:  'opss-data-source-1, opss-data-source-2'
+              AlgorithmType:  'FailOver'
+           ....
+```
+
+You can run WDT tool `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCDataSourceParams` for the complete list
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:384]: re-write: For the complete list of fields, run the WDT command, `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParmas`.

**Review by rosemarymarano — COMMENTED** [review]
Edits

File: documentation/2.0/content/rcuinfo.md [hunk 16]
```diff
@@ -325,19 +325,23 @@ you can specify additional `JDBCOracleParams` under `JdbcResource` of each data
                 ....
 ```
 
-You can run WDT tool `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParmas` for the complete list 
-of fields.
+For the complete list of fields, run the WDT command, 
+
+```shell
+modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParams
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:331]: Preceded commands with a dollar sign ($)
$ modelHelp.sh

File: documentation/2.0/content/rcuinfo.md [hunk 17]
```diff
@@ -381,6 +385,10 @@ For example,
            ....
 ```
 
-You can run WDT tool `modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCDataSourceParams` for the complete list
-of fields.
+
+For the complete list of fields, run the WDT command,
+
+```shell
+modelHelp.sh -oracle_home <oracle home> resources:/JDBCSystemResource/JdbcResource/JDBCOracleParams
```
Comment (rosemarymarano) [inline documentation/2.0/content/rcuinfo.md:392]: $ modelHelp.sh

**Review by rosemarymarano — APPROVED** [review]
Two minor fixes otherwise ok.

**Review by robertpatrick — APPROVED** [review]

**Review by CarolynRountree — APPROVED** [review]

**Review by rakillen — APPROVED** [review]

**Review by ddsharpe — APPROVED** [review]

**Pull request merged**